sample-root-scripts/read_LIGen_OD_output.C

```diff
@@ -0,0 +1,335 @@
+// C++ Includes
+#include <iostream>
+//#include <algorithm>
+//#include <iomanip>
+#include <cmath>
+
+// ROOT Includes
+#include "TH1F.h"
+#include "TH2F.h"
+#include "TFile.h"
+//#include "TApplication.h"
+
+
+// WCSim Includes
+#include "WCSimRootEvent.hh"
+#include "WCSimRootGeom.hh"
+
+// Defines
+#define PI 3.141592654
+#define topPMT 19
+/*
+*How to run:
+*
+* enter in the terminal $WCSIMDIR/rootwc/rootwc -l 'read_OD.C("WCSim.root","outputFile.root",false)' to run the code
+* where you replace WCSim.root with your file name and outputfile with the name you wish to save it under
+*/
+
+// A function to easily format the histograms
+void format_histogram(TH1D* h, std::string title ,std::string xtitle, std::string ytitle){
+	h->SetTitle( title.c_str() );
+	h->GetXaxis()->SetTitle( xtitle.c_str() );
+	h->GetYaxis()->SetTitle( ytitle.c_str() );
+	h->SetLineColor(kRed);
+	h->SetLineWidth(2);
+}
+// An (overloaded) function to easily format the histograms
+void format_histogram(TH2D* h, std::string title ,std::string xtitle, std::string ytitle){
+	h->SetTitle( title.c_str() );
+	h->GetXaxis()->SetTitle( xtitle.c_str() );
+	h->GetYaxis()->SetTitle( ytitle.c_str() );
+}
+
+void read_LIGen_OD_output(std::string inFileName, std::string outFileName, bool verbosity=0){
+
+
+	// Some nicely formatted text options
+	std::cout << std::scientific; // This causes all numbers to be displayed in scientific notation.
+	std::cout << std::setprecision(2); // Sets the decimal precision (no more than two decimal places)
+	std::cout << std::left; // Sets the text justification to left
+	const int txtW = 20; // Width of "box" holding text
+	const int numW = 10; // Width of "box" holding numbers
+
+
+	// Open the WCSim file
+	TFile *inFile = new TFile(inFileName.c_str(), "READ");
+	if ( !inFile->IsOpen() ){
+		std::cout << "Error: could not open input file \"" << inFileName << "\"." <<std::endl;
+
+	} else if (verbosity) {
+		std::cout << "Input file: " << inFileName << std::endl;
+	}
+
+
+
+	// Get a pointer to the tree from the input file
+	TTree *wcsimTree = (TTree*) inFile->Get("wcsimT");
+
+	// Get the number of events in the tree
+	long int nEvent = wcsimTree->GetEntries();
+	if (verbosity) { std::cout << "Number of events: "<< nEvent << std::endl;}
+
+	// Create a WCSimRootEvent to put stuff from the tree in
+	WCSimRootEvent *wcsimRootID = new WCSimRootEvent();
```

Comment:
Sorry to ask, but will be useful when ID LI's are starting to be used
Can you add the mPMT stuff too? This is saved in a third branch `wcsimrootevent2`

Reply:
Yeah that's great thanks